application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubCommand.java

```diff
@@ -0,0 +1,118 @@
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.utils.StringDistances;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.TemporalAmount;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.stream.Stream;
+
+/**
+ * Slash command (/github) used to search for an issue in one of the repositories listed in the
+ * config. It even comes with auto completion!
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class GitHubCommand extends SlashCommandAdapter {
+    private static final long ONE_MINUTE_IN_MILLIS = 60_000L;
+
+    /**
+     * Compares the getUpdatedAt values
+     */
+    private static final Comparator<GHIssue> GITHUB_ISSUE_TIME_COMPARATOR = (i1, i2) -> {
+        try {
+            return i2.getUpdatedAt().compareTo(i1.getUpdatedAt());
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    };
+
+    private static final String TITLE_OPTION = "title";
+
+    private final GitHubReference reference;
+
+    private Instant lastCacheUpdate;
+    private List<String> autocompleteCache;
+
+    public GitHubCommand(GitHubReference reference) {
+        super("github-search", "Search configured GitHub repositories for an issue/pull request",
+                CommandVisibility.GUILD);
+
+        this.reference = reference;
+
+        getData().addOption(OptionType.STRING, TITLE_OPTION,
+                "Title of the issue you're looking for", true, true);
+
+        updateCache();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        String title = event.getOption(TITLE_OPTION).getAsString();
+        Matcher matcher = GitHubReference.ISSUE_REFERENCE_PATTERN.matcher(title);
+
+        if (!matcher.find()) {
+            event.reply(
+                    "Could not parse your query. Was not able to find an issue number in it (e.g. #207).")
+                .setEphemeral(true)
+                .queue();
+
+            return;
+        }
+
+        reference.findIssue(Integer.parseInt(matcher.group("id")))
+            .ifPresentOrElse(issue -> event.replyEmbeds(reference.generateReply(issue)).queue(),
+                    () -> event.reply("Could not find the issue you are looking for.")
+                        .setEphemeral(true)
+                        .queue());
+    }
+
+    @Override
+    public void onAutoComplete(CommandAutoCompleteInteractionEvent event) {
+        String title = event.getOption(TITLE_OPTION).getAsString();
+
+        if (title.isEmpty()) {
+            event.replyChoiceStrings(autocompleteCache.stream().limit(25).toList()).queue();
+        } else {
+            Queue<String> queue = new PriorityQueue<>(Comparator.comparingInt(
```

Comment:
the name isnt really optimal. maybe `closestMatches` or `closestSuggestions`.

it also reads a bit complex since u nested everything. i would at least move out the 25-list into a variable:
```java
Queue<String> closestMatches = ...
List<String> suggestions = ...

event.replyChoiceStrings(suggestions).queue();
```
maybe also the logic u use to compute the score. So that its `Comparator.comparingInt(this::scoreSuggestion)` or similar.

Reply:
done

Reply:
could u also move the `Stream.generate(closestSuggestions::poll).limit(25).toList()` into its own variable? Its a bit harder to read when its all packed together with the JDA stuff of message-sending.